flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,413 @@
+        assert isinstance(dc["r"], list), '"r" dataset is not a list'
+        assert isinstance(dc["g"], list), '"g" dataset is not a list'
+        assert isinstance(dc["b"], list), '"b" dataset is not a list'
+        return dc
+
+
+def check_deserialize(response):
+    dc_type = response["dataContainer"]["type"]
+    match dc_type:
+        case "OrderedPair":
+            OrderedPairModel.parse_obj(response)
+        case "OrderedTriple":
+            OrderedTripleModel.parse_obj(response)
+        case "DataFrame":
+            DataFrameModel.parse_obj(response)
+        case "Matrix":
+            MatrixModel.parse_obj(response)
+        case "Grayscale":
+            GrayscaleModel.parse_obj(response)
+        case "Scalar":
+            ScalarModel.parse_obj(response)
+        case "Vector":
+            ScalarModel.parse_obj(response)
+        case "Image":
+            ImageModel.parse_obj(response)
+        case _:
+            raise TypeError(
+                f"Unsupported DataContainer type: {dc_type}. Check case (e.g. OrderedPair)."
+            )
+
+
+class FlojoyCloud:
+    """
+    A class that allows pulling and pushing DataContainers from the
+    Flojoy cloud client (cloud.flojoy.ai).
+
+    Returns data in a pythonic format (e.g. Pillow for images,
+    DataFrames for arrays/matrices).
+
+    Will support the majority of the Flojoy cloud API:
+    https://rest.flojoy.ai/api-reference
+
+    Recommended for api key:
+    utils.get_credentials()[0]["value"]
+    or
+    os.environ.get("FLOJOY_CLOUD_KEY")
+    """
+
+    def __init__(self, apikey):
+        self.headers = {"api_key": apikey}
+        self.valid_types = [
+            "OrderedPair",
+            "OrderedTriple",
+            "DataFrame",
+            "Grayscale",
+            "Matrix",
+            "Scalar",
+            "Vector",
+            "Image",
+        ]
+
+    def create_payload(self, data, dc_type):
+        """
+        A method that formats data into a payload that can be handled by
+        the Flojoy cloud client.
+        """
+        assert (
+            dc_type in self.valid_types
+        ), f"Type {dc_type} not supported. Check capitals (e.g. OrderedPair)."
+        match dc_type:
+            case "OrderedPair":
+                if isinstance(data, dict) and "x" in data:
+                    payload = json.dumps(
+                        {
+                            "data": {
+                                "type": "OrderedPair",
+                                "x": data["x"],
+                                "y": data["y"],
+                            }
+                        },
+                        cls=NumpyEncoder,
+                    )
+                else:
+                    print(
+                        "For ordered pair type, data must be in"
+                        " dictionary form with keys 'x' and 'y'"
+                    )
+                    raise TypeError
```

Comment:
Something like this might be better (and similarly for ordered triple)
```suggestion
                if not (isinstance(data, dict) and "x" in data and "y" in data):
                    raise TypeError("For ordered pair type, data must be in dictionary form with keys 'x' and 'y'")
                payload = json.dumps(
                    {
                        "data": {
                            "type": "OrderedPair",
                            "x": data["x"],
                            "y": data["y"],
                        }
                    },
                    cls=NumpyEncoder,
                )
```